api/organisations/tasks.py

```diff
@@ -189,7 +201,88 @@ def handle_api_usage_notifications():
             )
 
 
+def charge_for_api_call_count_overages():
+    now = timezone.now()
+
+    # Get the period where we're interested in any new API usage
+    # notifications for the relevant billing period (ie, this month).
+    api_usage_notified_at = now - timedelta(days=30)
+
+    # Since we're interested in monthly billed accounts, set a wide
+    # threshold to catch as many billing periods that could be roughly
+    # considered to be a "monthly" subscription.
+    month_window_start = timedelta(days=25)
+    month_window_end = timedelta(days=35)
+
+    # Only apply charges to ongoing subscriptions that are close to
+    # being charged due to being at the end of the billing term.
+    closing_billing_term = now + timedelta(hours=12)
+
+    organisation_ids = set(
+        OrganisationAPIUsageNotification.objects.filter(
+            notified_at__gte=api_usage_notified_at,
+            percent_usage__gte=100,
+        )
+        .exclude(
+            organisation__api_billing__billed_at__gt=api_usage_notified_at,
+        )
+        .values_list("organisation_id", flat=True)
+    )
+
+    for organisation in Organisation.objects.filter(
+        id__in=organisation_ids,
+        subscription_information_cache__current_billing_term_ends_at__lte=closing_billing_term,
+        subscription_information_cache__current_billing_term_ends_at__gte=now,
```

Comment:
Well we can do it after the billing period is over, but it just feels better to bill them within the same period. It's how my cell phone bill works, after all. Plus we get to collect the overage a little sooner.

One thing we can consider is to run the task more frequently and bill them, say, one hour before the subscription is billed. 

Or we can do it your way and push it into the next billing period. I'm not totally opposed to that.

Reply:
> Well we can do it after the billing period is over, but it just feels better to bill them within the same period. It's how my cell phone bill works, after all. Plus we get to collect the overage a little sooner.

Yeah, this would be the ideal, but I think for e.g. your cell phone bill there's a window of time between when your billing period ends, and when your bill is calculated. If there's a way we can do that, then great, otherwise maybe this is the best option: 

> One thing we can consider is to run the task more frequently and bill them, say, one hour before the subscription is billed.

Reply:
I've read through the documentation and I don't think there is a way of separating the end of the subscription term and the application of addons, so I'm going to run the billing more frequently. I think it's ok to have unbilled API usage at the last ~30 minutes of a month.

Reply:
Ok this has been added.